r-package/grf/tests/testthat/test_forest_summaries.R

```diff
@@ -138,3 +138,33 @@ test_that("causal forest calibration works with clusters", {
 
   expect_equal(tc[, 1:3], tc.clust[, 1:3])
 })
+
+test_that("best linear projection is reasonable", {
+  n <- 2000
+  p <- 5
+  X <- matrix(rnorm(n * p), n, p)
+  W <- rbinom(n, 1, 0.25 + 0.5 * (X[, 1] > 0))
+  S <- rbinom(n, 1, 0.5)
+  Y <- S * pmax(X[, 1], 0) * W + X[, 2] + pmin(X[, 3], 0) + rnorm(n)
+  forest <- causal_forest(X, Y, W, num.trees = 300)
+  
+  blp.all <- best_linear_projection(forest, X[,1:2])
+  expect_equal(blp.all[1,1], 0.4/2, tol = 0.1)
+  expect_equal(blp.all[2,1], 0.5/2, tol = 0.1)
+  expect_equal(blp.all[3,1], 0.0/2, tol = 0.1)
+  
```

Comment:
It's a little opaque using the literal constants .4 and .5 for the regression coefficients of Z * 1(Z > 0) on 1 and Z for standard normal Z.  A comment, or just approximating those coefficients here, would improve readability.